### DISCUSSION TITLE
Error: ** Note: you can use --noprofile to disable default.profile **

### DISCUSSION THREAD

**Discussion by librewolffy** [netblue30/firejail#5256] [Q&A]
CLI: firejail appname

Getting errors:
Reading profile /etc/firejail/default.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/default-program.inc

** Note: you can use --noprofile to disable default.profile **

And then the information error shows:
"Please run ... as root with sudo or install gksu / kdesudo"

This error shows from firetools as well.

The app won't start and this error shows instead:
Please run ... as root with sudo or install gksu / kdesudo

What is the solution for this? I think the problem is the profile; could it be the profile problem per the error message?

Does anyone else know a solution for this error?

**Comment by rusty-snake** [comment]
FTR: https://github.com/netblue30/firetools/issues/65

**Comment by librewolffy** [comment]
No one is answering or providing the solution over there, so I thought posting it here to see if anyone has had the same problem. I provided more details, Do you know how to resolve this? I want to use firejail with the app at least from the CLI. I would prefer to add it to firetools.

**Comment by reinerh** [comment]
> Please run ... as root with sudo or install gksu / kdesudo

The only place where I could find this error message was in a stackoverflow post:
https://stackoverflow.com/questions/5747007/get-root-dialog-in-python-on-mac-os-x-windows

The script there checks if either the user is root, or if one of gksudo/kdesu/pkexec is available.
So it sounds like your application really wants to run as root. Either by starting directly as root, or by using some other tool to escalate your privileges.

You need to give more details about the program you want to start. It sounds like it is hard to isolate if it requires running as root.

**Comment by gitsteff** [comment]
The default.profile sets the option
`nonewprivs`
which means that no privileg escalation should be possible.
Try to uncomment in the default.profile and see if it helps, but do not leave it uncommented. If it works apply a specific profile for this app only with nonewprivs commented out, do not leave it as default.
HTH.

**Reply by rusty-snake** [reply]
FTR: You can "comment" it on the command line with `--ignore=nonewprivs`.

**Reply by rusty-snake** [reply]
The default profile also contains
- `seccomp` and `procotol` with imply NNP.
- `caps.drop all` which zero-out the capability bounding set.
- `noroot` which "removes" the root user.

**Comment by librewolffy** [comment]
The option is uncommented as default as is.

I commented it out not to leave it uncommented and still the same error: "Please run ... as root with sudo or install gksu / kdesudo"

I have just tried "firejail --ignore=nonewprivs appname", still the same error.

For this CLI "sudo firejail --ignore=nonewprivs appname" or "sudo firejail appname", the message error as follows:
Apppname is not able to write its preferences.
Please check the writable permissions of your config file: /home/user/.config/..../app.conf
Do you want Appname try to fix it?

When I select the "No", the app starts. There is no message error: "Please run ... as root with sudo or install gksu / kdesudo"

It seems there is a problem with the app config file, but I don't want to modify anything to it. It starts without using firejail from CLI. 

I would prefer to do something on the firejail or firetools sides. 

I don't know much, so does the app message error say anything to you? 

The app starts when selecting "No" but what does that app message error precisely mean?

**Reply by reinerh** [reply]
As already mentioned, without more details about the application it's hard to analyze the problem.

**Reply by gitsteff** [reply]
Try running your cli command with firejail option --debug. This gave me some hints about how firejail is working and what happens during startup. Also there are infos about rights escalation and dropping maybe this could help.

**Comment by librewolffy** [comment]
Would firejail or firetools require such specifics in order to start the app?

**Reply by reinerh** [reply]
That doesn't really help. Can you link the source code to it? It's interesting how it's started. Is there a script that starts it, is it a suid binary, ...?
